packages/amplify-graphql-relational-transformer/src/utils.ts

```diff
@@ -254,6 +279,7 @@ export const getReferencesNodes = (
     const fieldNode = relatedType.fields!.find((field) => field.name.value === fieldName);
 
     if (!fieldNode) {
+      // TODO: include more directive / type information to help debugging.
```

Comment:
After going through the error message assertions in our unit tests, there's a lot of room for improvement.
Tracking this work in feature branch PR as:
> Improve error messages surfaced by including more directive and type information to improve debugging.